providers/async_base.py

```
@@ -0,0 +1,545 @@
+            prompt (str): The user prompt to send to the model.
+            model_name (str): The name of the model to use.
+            system_prompt (Optional[str]): An optional system prompt to guide model behavior.
+            temperature (float): Sampling temperature for response variability.
+            max_output_tokens (Optional[int]): Maximum number of tokens to generate.
+
+        Returns:
+            ModelResponse: The generated content and associated metadata.
+        """
+        semaphore = await self.get_semaphore()
+        async with semaphore:
+            logger.debug(f"Acquired semaphore for {model_name} request")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)